### PR TITLE
First working reduction implemented

### PR DESCRIPTION
**Pull request by bprzybylski**

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -31,25 +32,44 @@ end
...
+    if force
         sources = joinpath(download_dir, "kbmag-$version.tar.gz")
         getsources(src_uri, sources, force)
-        unpack(sources, src_dir, force)
+        unpack(sources, sources_dir, force)
+    end
+
+    standalone_lib_dir      = joinpath(sources_dir, "kbmag-$version", "standalone", "lib")
+    standalone_sources_dir  = joinpath(sources_dir, "kbmag-$version", "standalone", "src")
+    standalone_bin_dir      = joinpath(sources_dir, "kbmag-$version", "standalone", "bin");   mkpath(standalone_bin_dir)
```
Comment (kalmarek) [inline deps/build.jl:51]: new line missing?

File: deps/deps.jl [hunk 2]
```diff
@@ -9,4 +8,21 @@ function check_deps(lib)
     end
 end
 
-check_deps(fsalib)
```
Comment (kalmarek) [inline deps/deps.jl:12]: shouldn't we `check_lib(fsalib)` ??

File: src/debug.jl [hunk 3]
```diff
@@ -20,3 +20,26 @@ function Base.show(io::IO, kbt::KBType)
...
+
+function Base.show(io::IO, re::ReductionEquation)
+    lhs, rhs = unsafe_load_ptrGen(re.lhs), unsafe_load_ptrGen(re.rhs)
+    println(io, "ReductionEquation (in rws generators):")
+    println(io, "\t$lhs  → $rhs")
+end
+
+# In the following, we ommit the first element of rws.eqns as it is garbage
+# (kbmag does not use array elements indexed with zero)
+eqns(rws::RewritingSystem) = [unsafe_load(rws.eqns, i) for i in 2:rws.num_eqns+1]
```
Comment (kalmarek) [inline src/debug.jl]: this should go to `RWS.jl`

File: src/reduce.jl [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                 Cint,
+                 (Ptr{Gen}, Ref{ReductionStruct}),
+                 w, rs)
+
+    iszero(r) || error(
+        "Call to rws_reduce in $fsalib returned non-zero output: $r")
+    return w
+end
+
+function gen_names(rws::RewritingSystem)
```
Comment (kalmarek) [inline src/reduce.jl]: this should go to `RWS.jl`

File: tmp.jl [hunk 5]
```diff
@@ -1,55 +1,50 @@
 using Pkg
 Pkg.activate(".")
 # Pkg.build()
-
```
Comment (kalmarek) [inline tmp.jl:4]: when merging to master we could just remove `tmp.jl`

**Review by kalmarek — CHANGES_REQUESTED** [review]

File: src/KBmag.jl [hunk 6]
```diff
@@ -2,15 +2,18 @@ module KBmag
...
 end
 
-include("types.jl")
-include("RWS.jl")
+include("types.jl")     # struct types that reflect original structs from fsalib
+include("RWS.jl")       # loading/saving and preparing rewriting systems
+include("reduce.jl")    # word reduction
 
-include("debug.jl")
+include("debug.jl")     # independent debug functions (ready to be removed at any moment)
```
Comment (kalmarek) [inline src/KBmag.jl:17]: update after renaming;

also: can you add the `BinWrapper.jl` if you have it lying around?

**Comment by kalmarek** [comment]
don't clean the history, we'll squash it anyway;

**Comment by kalmarek** [comment]
also add `.travis.yml` to main directory with the following content:
```yml
# Documentation: http://docs.travis-ci.com/user/languages/julia/
language: julia
os:
  - linux
  - osx
julia:
  - 1.0
  - 1.1
  - 1.2
  - nightly
notifications:
  email: false
matrix:
  fast_finish: true
  allow_failures:
    - julia: nightly
codecov: true
```
and enable Travis CI

File: deps/build.jl (hunk 1 above)
Comment (bprzybylski) [reply deps/build.jl:51]: I wanted to stay consistent here, see `deps/build.jl:4-8`

File: deps/deps.jl (hunk 2 above)
Comment (bprzybylski) [reply deps/deps.jl:12]: Yes, we do that in line 22.

File: src/debug.jl (hunk 3 above)
Comment (bprzybylski) [reply src/debug.jl]: I moved that definition to `RWS.jl`.

File: src/reduce.jl (hunk 4 above)
Comment (bprzybylski) [reply src/reduce.jl]: I moved that definition to `RWS.jl`.

File: src/KBmag.jl (hunk 6 above)
Comment (bprzybylski) [reply src/KBmag.jl:17]: BinWrapper.jl is not ready yet.

File: tmp.jl (hunk 5 above)
Comment (bprzybylski) [reply tmp.jl:4]: I am not sure whether to remove it or ommit it while merging.

File: deps/build.jl (hunk 1 above)
Comment (kalmarek) [reply deps/build.jl:51]: ok!

File: tmp.jl (hunk 5 above)
Comment (kalmarek) [reply tmp.jl:4]: if we were very picky, we shoud `git filter-branch`, but for here just ommit it

File: deps/deps.jl (hunk 2 above)
Comment (kalmarek) [reply deps/deps.jl:12]: ok, I'm not used to read multiple calls per line ;-)

**Review by kalmarek — APPROVED** [review]

**Comment by kalmarek** [comment]
but tmp.jl is still committed

**Pull request merged**